src/device.rs

```diff
@@ -0,0 +1,347 @@
+                    }
+                }
+                DeviceType::Xcb => {
+                    #[cfg(feature = "xcb")]
+                    {
+                        ffi::cairo_xcb_device_debug_cap_xrender_version(self.to_raw_none(),
+                                                                        major_version,
+                                                                        minor_version)
+                    }
+                    #[cfg(not(feature = "xcb"))]
+                    {
+                        panic!("you need to enable \"xcb\" feature")
+                    }
+                }
+                _ => {
+                    panic!("invalid device type")
+                }
+            }
+        }
+    }
+
+    #[cfg(any(feature = "xlib", feature = "xcb", feature = "dox"))]
+    pub fn debug_get_precision(&self) -> i32 {
+        unsafe {
+            match self.get_type() {
+                DeviceType::Xlib => {
+                    #[cfg(feature = "xlib")]
+                    {
+                        ffi::cairo_xlib_device_debug_get_precision(self.to_raw_none())
+                    }
+                    #[cfg(not(feature = "xlib"))]
+                    {
+                        panic!("you need to enable \"xlib\" feature")
+                    }
+                }
+                DeviceType::Xcb => {
+                    #[cfg(feature = "xcb")]
+                    {
+                        ffi::cairo_xcb_device_debug_get_precision(self.to_raw_none())
+                    }
+                    #[cfg(not(feature = "xcb"))]
+                    {
+                        panic!("you need to enable \"xcb\" feature")
+                    }
+                }
+                _ => {
+                    panic!("invalid device type")
+                }
+            }
+        }
+    }
+
+    #[cfg(any(feature = "xlib", feature = "xcb", feature = "dox"))]
+    pub fn debug_set_precision(&self, precision: i32) {
+        unsafe {
+            match self.get_type() {
+                DeviceType::Xlib => {
+                    #[cfg(feature = "xlib")]
+                    {
+                        ffi::cairo_xlib_device_debug_set_precision(self.to_raw_none(), precision)
+                    }
+                    #[cfg(not(feature = "xlib"))]
+                    {
+                        panic!("you need to enable \"xlib\" feature")
+                    }
+                }
+                DeviceType::Xcb => {
+                    #[cfg(feature = "xcb")]
+                    {
+                        ffi::cairo_xcb_device_debug_set_precision(self.to_raw_none(), precision)
+                    }
+                    #[cfg(not(feature = "xcb"))]
+                    {
+                        panic!("you need to enable \"xcb\" feature")
+                    }
+                }
+                _ => {
+                    panic!("invalid device type")
```

Comment:
Maybe here (and elsewhere) you want to print the actual device type in the panic?